<a href="https://colab.research.google.com/github/Oseiasdfarias/ProbabilidadeProcessosEstocarticos/blob/main/FEE_PPE_LE_U04_P01_Oseias_Dias_de_Farias_201733940002.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <center>Lista de Exemplos U04-Parte 01</center>

<br>

Aluno: Oséias Dias de Farias

Matrícula: 201733940002

<br>

## <center>4.1 – Conceito de Variável Aleatória (V.A.) e</center> 
## <center>4.2 – Probabilidade associada à variável aleatória (V.A.)</center>

<br>


### **Questão 1** – Um experimento $(E)$ consiste em jogar uma moeda 4 vezes.


**A)** Especifique o espaço amostral $(S)$, onde $C$ corresponde a “cara"e $K$ corresponde a “coroa".

<br>

+ RESPOSTA

<br>

S = [ 
  
  KKKK, KKKC, KKCK, KKCC,
  
  KCKK, KCCK, KCKC, KCCC,
  
  CCCC, CCCK, CCKC, CCKK,
  
  CKCC,CKCK, CKKC, CKKK
  
  ]

<br><br>

**B)** Seja a Variável Aleatória $(X)$ a ocorrencia de “coroas” nas 4 jogadas. Especifique os resultados de $S$, os valores de $X$ correspondentes (contradomínio $R_x$) e a probabilidade de cada valor de $X$.

<br>

+ RESPOSTA

<img src="https://github.com/Oseiasdfarias/ProbabilidadeProcessosEstocarticos/blob/main/LISTA_04_P1/Q1_B.png?raw=true" alt="some text" width=70% height=70%>

<br>

### **Questão 2** – Um experimento $(E)$ consiste em jogar 3 dados (de 6 faces).


**A)** Especifique o espaço amostral $(S)$. Especificando a ocorrencia das faces pelo número correspondente (1, 2, 3, 4, 5 ou 6).


<br>

+ RESPOSTA

E = [

(000),  (001),  (002),  (003),  (004),  (005),  (006),  
(010),  (011),  (012),  (013),  (014),  (015),  (016),  
(020),  (021),  (022),  (023),  (024),  (025),  (026),  
(030),  (031),  (032),  (033),  (034),  (035),  (036),  
(040),  (041),  (042),  (043),  (044),  (045),  (046),  
(050),  (051),  (052),  (053),  (054),  (055),  (056),  
(060),  (061),  (062),  (063),  (064),  (065),  (066),  


(100),  (101),  (102),  (103),  (104),  (105),  (106),  
(110),  (111),  (112),  (113),  (114),  (115),  (116),  
(120),  (121),  (122),  (123),  (124),  (125),  (126),  
(130),  (131),  (132),  (133),  (134),  (135),  (136),  
(140),  (141),  (142),  (143),  (144),  (145),  (146),  
(150),  (151),  (152),  (153),  (154),  (155),  (156),  
(160),  (161),  (162),  (163),  (164),  (165),  (166),  


(200),  (201),  (202),  (203),  (204),  (205),  (206),  
(210),  (211),  (212),  (213),  (214),  (215),  (216),  
(220),  (221),  (222),  (223),  (224),  (225),  (226),  
(230),  (231),  (232),  (233),  (234),  (235),  (236),  
(240),  (241),  (242),  (243),  (244),  (245),  (246),  
(250),  (251),  (252),  (253),  (254),  (255),  (256),  
(260),  (261),  (262),  (263),  (264),  (265),  (266),  


(300),  (301),  (302),  (303),  (304),  (305),  (306),  
(310),  (311),  (312),  (313),  (314),  (315),  (316),  
(320),  (321),  (322),  (323),  (324),  (325),  (326),  
(330),  (331),  (332),  (333),  (334),  (335),  (336),  
(340),  (341),  (342),  (343),  (344),  (345),  (346),  
(350),  (351),  (352),  (353),  (354),  (355),  (356),  
(360),  (361),  (362),  (363),  (364),  (365),  (366),  


(400),  (401),  (402),  (403),  (404),  (405),  (406),  
(410),  (411),  (412),  (413),  (414),  (415),  (416),  
(420),  (421),  (422),  (423),  (424),  (425),  (426),  
(430),  (431),  (432),  (433),  (434),  (435),  (436),  
(440),  (441),  (442),  (443),  (444),  (445),  (446),  
(450),  (451),  (452),  (453),  (454),  (455),  (456),  
(460),  (461),  (462),  (463),  (464),  (465),  (466),  


(500),  (501),  (502),  (503),  (504),  (505),  (506),  
(510),  (511),  (512),  (513),  (514),  (515),  (516),  
(520),  (521),  (522),  (523),  (524),  (525),  (526),  
(530),  (531),  (532),  (533),  (534),  (535),  (536),  
(540),  (541),  (542),  (543),  (544),  (545),  (546),  
(550),  (551),  (552),  (553),  (554),  (555),  (556),  
(560),  (561),  (562),  (563),  (564),  (565),  (566),  


(600),  (601),  (602),  (603),  (604),  (605),  (606),  
(610),  (611),  (612),  (613),  (614),  (615),  (616),  
(620),  (621),  (622),  (623),  (624),  (625),  (626),  
(630),  (631),  (632),  (633),  (634),  (635),  (636),  
(640),  (641),  (642),  (643),  (644),  (645),  (646),  
(650),  (651),  (652),  (653),  (654),  (655),  (656),  
(660),  (661),  (662),  (663),  (664),  (665),  (666) 

]

In [ ]:
print("E = [")
for i in range(7):
  for j in range(7):
    for n in range(7):
      print(f"({i}{j}{n}), ", end=" ")
    print()
  print("\n")
print("]")

**B)** Seja a Variável Aleatória $(X)$ a soma dos valores das duas primeiras faces menos o valor da terceira. Especifique os resultados de $S$, os valores de $X$ correspondentes (contradomínio $R_x$) e a probabilidade de cada valor de $X$.


<br>

+ RESPOSTA

**Contradomínio $R_x$**

Rx = [0, -1, -2, -3, -4, -5, -6, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]

<br>

**Probabilidade de cada valor de $X$**

$X_0 = 0.00291545 , X_1 = 0.00874636, X_2 = 0.01749271$

$X_3 = 0.02915452, X_4 = 0.04373178, X_5 = 0.06122449$

$X_6 = 0.08163265, X_7 = 0.09620991, X_8 = 0.10495627$

$X_9 = 0.10787172, X_{10} = 0.10495627, X_{11} = 0.09620991$

$X_{12} = 0.08163265, X_{13} = 0.06122449, X_{14} = 0.04373178$

$X_{15} = 0.02915452, X_{16} = 0.01749271, X_{17} = 0.00874636$

$X_{18} = 0.00291545$

<br>

**Resultado de $S$**

In [ ]:
s = []
for i in range(7):
  for j in range(7):
    for n in range(7):
      soma = (i+j)-n
      s.append(soma)
      print(f"x({i}{j}{n}) = {soma}, ", end=" ")
    print()
  print("\n")

unicos_lista = list(dict.fromkeys(s))
print("Rx", unicos_lista)

**Encontrando o Contradomínio $R_x$ e as Probabilidades de cada valor de $X$**

In [ ]:
import numpy as np

prob = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype="float64")
for i in s:
  if (i == -6):
    prob[0] += 1/len(s)
  elif (i == -5):
    prob[1] += 1/len(s)
  elif (i == -4):
    prob[2] += 1/len(s)
  elif (i == -3):
    prob[3] += 1/len(s)
  elif (i == -2):
    prob[4] += 1/len(s)
  elif (i == -1):
    prob[5] += 1/len(s)
  elif (i == 0):
    prob[6] += 1/len(s)
  elif (i == 1):
    prob[7] += 1/len(s)
  elif (i == 2):
    prob[8] += 1/len(s)
  elif (i == 3):
    prob[9] += 1/len(s)
  elif (i == 4):
    prob[10] += 1/len(s)
  elif (i == 5):
    prob[11] += 1/len(s)
  elif (i == 6):
    prob[12] += 1/len(s)
  elif (i == 7):
    prob[13] += 1/len(s)
  elif (i == 8):
    prob[14] += 1/len(s)
  elif (i == 9):
    prob[15] += 1/len(s)
  elif (i == 10):
    prob[16] += 1/len(s)
  elif (i == 11):
    prob[17] += 1/len(s)
  elif (i == 12):
    prob[18] += 1/len(s)

print(f"\nProbabilidade para os Xs =\n {prob}\n")
print(f"Soma das Probabilidades: {prob.sum()}")


Probabilidade para os Xs =
 [0.00291545 0.00874636 0.01749271 0.02915452 0.04373178 0.06122449
 0.08163265 0.09620991 0.10495627 0.10787172 0.10495627 0.09620991
 0.08163265 0.06122449 0.04373178 0.02915452 0.01749271 0.00874636
 0.00291545]

Soma das Probabilidades: 1.0000000000000004


<br>

## <center>4.3 – Variáveis aleatórias discretas e contínuas e </center>
## <center>4.4 – Funções de variáveis aleatórias $(V.A.)$ – fdp e FDP</center>


### **Questão 3** – Com base na **Questão 1**. Determine:


**A)** $p(x_i)-$fdp de $X$.


<br>

+ RESPOSTA

<br>

|x_i 	     | 0 	  | 1 	| 2   	| 3 	| 4 	  |
|-----	   |---	  |---	|---	  |---	|---	  |
|P[X = x_i]| 6.25% |25%   |  37.5% | 25%  |  6.25% |


<br>

**B)** $F(x_i)-$FDP de $X$.


<br>

+ RESPOSTA

|x_i 	     | 0 	  | 1 	| 2   	| 3 	| 4 	  |
|-----	   |---	  |---	|---	  |---	|---	  |
|F[X = x_i]| 6.25% |31.25%   |  68.75% | 93.75%  |  100.0% |

<br>

In [109]:
prob_q1 = np.array([1/16, 1/4, 3/8, 1/4, 1/16], dtype="float64")
fdp_q1 = prob_q1*100.0

print(f"\np(xi)− fdp de X:")
for i in fdp_q1:
  print(f"{i}% ", end=" ")

FDP_q1 = np.zeros(5)
for i in range(0, 5):
  FDP_q1[i] = fdp_q1[:i+1].sum()

print(f"\n\nF(xi)−FDP de  X:")
for i in FDP_q1:
  print(f"{i}% ", end=" ")



p(xi)− fdp de X:
6.25%  25.0%  37.5%  25.0%  6.25%  

F(xi)−FDP de  X:
6.25%  31.25%  68.75%  93.75%  100.0%  

### **Questão 4** – Com base na **Questão 2**. Determine:


a) $p(x_i)-$fdp de $X$.


<br>

+ RESPOSTA

<br>

| x_i| 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 |
|------|---|---|---|---|---|---|---|---|---|--- |
|P[X = x_i] |0.29\% |  0.87\% | 1.74\% | 2.91\% | 4.37\% | 6.12\% | 8.16\% | 9.62\% | 10.49\%| 10.78\% |
|x_i | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18|
|P[X = x_i] | 10.49\% | 9,62\% | 8.16\% | 6.12\% | 4.37\% | 2.91\% | 1.74\% | 0.87\% | 0.29\%|

<br>

In [132]:
fdp_q2 = prob*100

print(f"\n\np(xi)− fdp de  X:")
for i in range(19):
  print(f"{format(fdp_q2[i], '.2f')}% ", end=" ")
  if (i == 9):
    print()



p(xi)− fdp de  X:
0.29%  0.87%  1.75%  2.92%  4.37%  6.12%  8.16%  9.62%  10.50%  10.79%  
10.50%  9.62%  8.16%  6.12%  4.37%  2.92%  1.75%  0.87%  0.29%  

b) $F(x_i)-$FDP de $X$.


<br>

+ RESPOSTA

In [131]:
FDP_q2 = np.zeros(19)
for i in range(0, 19):
  FDP_q2[i] = fdp_q2[:i+1].sum()

print("\nF(xi)− FDP de  X: ")
for i in range(19):
  print(f"{format(FDP_q2[i], '.2f')}% " , end=" ")
  if (i == 9):
    print()



F(xi)− FDP de  X: 
0.29%  1.17%  2.92%  5.83%  10.20%  16.33%  24.49%  34.11%  44.61%  55.39%  
65.89%  75.51%  83.67%  89.80%  94.17%  97.08%  98.83%  99.71%  100.00%  

### **Questão 5** – Uma função distribuição de probabilidade acumulada FDP é definida da seguinte forma:

- $X < a \rightarrow F = 0;$
- $a \leq X \leq b \rightarrow F = \dfrac{x-a}{b-a};$
- $X > b \rightarrow F = 1;$

<br>

a) Calcule $f(x)$ - fdp de $X$.

<br>

+ RESPOSTA

<br>

b) Calcule $P[1 < X \leq 3]$ para $a = 1$ e $b = 5$.

<br>

+ RESPOSTA

<br>

c) Calcule $P[-1 < X \leq 2]$ para $a = 1$ e $b = 5$.

<br>

+ RESPOSTA

<br>

d) Calcule $P[- \infty < X \leq 1,5]$ para $a = 1$ e $b = 5$.

<br>

+ RESPOSTA

<br>

e) Calcule $P[0 < X \leq 6]$ para $a = 1$ e $b = 5$.

<br>

+ RESPOSTA

<br>

### **Questão 6** – O tempo de transmissão X de mensagens em um sistema de comunicação obedece a lei de probabilidade exponencial com parâmetro $λ$, isto é $P[X > x] = e −λx, x > 0$. Calcule, $T = 1/λ$.


a) Defina $F(x)$ - FDP de $X$


<br>

+ RESPOSTA


b) Calcule $f(x)$ - fdp de $X$.


<br>

+ RESPOSTA

c) Calcule $P[T < X \leq 2T]$ para $T = T = 1/λ.$


<br>

+ RESPOSTA
